File: src/ast/IntegerCastExpression.java
package ast;

import java.io.StringWriter;

public class IntegerCastExpression extends ParentNode {
    private String _number;

    public IntegerCastExpression(BaseNode type, String number) {
        super(NodeType.IntegerCastExpression, type);
        _number = number;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("(");
        child.Print(writer);
        writer.write(")");
        writer.write(_number);
    }
}

File: src/de/fearlesstobi/demangler/ast/IntegerCastExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class IntegerCastExpression(type: BaseNode?, private val number: String) : ParentNode(NodeType.IntegerCastExpression, type) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("(")
        child!!.print(writer)
        writer.write(")")
        writer.write(number)
    }

}